### PR TITLE
Remove deal & optimize firunsort

### PR DESCRIPTION
**Pull request by Marcos-cat**

### CODE REVIEW

File: src/optimize.rs [hunk 1]
```diff
@@ -57,6 +57,11 @@ pub(crate) fn optimize_instrs_mut(
             instrs.pop();
             instrs.push(Instr::ImplPrim(MemberOfRange, span))
         }
+        // First UnSort = random row
+        ([.., Instr::ImplPrim(UnSort, _)], Instr::Prim(First, span)) => {
```
Comment (kaikalii) [inline src/optimize.rs:61]: Should do `last` as well.

Comment (kaikalii) [reply src/optimize.rs:61]: `Instr::Prim(First | Last, span)`

Comment (Omnikar) [reply src/optimize.rs:61]: I think this would also require changing the error message at algorithm/dyadic/mod.rs:1943

Comment (kaikalii) [reply src/optimize.rs:61]: I'm not opposed to the error just saying something like "Can't pick random row from empty array."

Comment (Marcos-cat) [reply src/optimize.rs:61]: Oh my lord, you guys are replaying my exact conversation with myself from two minutes before I PRd this

Comment (Marcos-cat) [reply src/optimize.rs:61]: (I’ll add it)

**Pull request merged**